services/server/server.go

```diff
@@ -310,14 +310,16 @@ type Server struct {
 
 // ServeGRPC provides the containerd grpc APIs on the provided listener
 func (s *Server) ServeGRPC(l net.Listener) error {
-	if s.config.Metrics.GRPCHistogram {
+	if s.config.Metrics.GRPCHistogram && s.config.Metrics.Address != "" {
 		// enable grpc time histograms to measure rpc latencies
 		grpc_prometheus.EnableHandlingTimeHistogram()
 	}
 	// before we start serving the grpc API register the grpc_prometheus metrics
 	// handler.  This needs to be the last service registered so that it can collect
 	// metrics for every other service
-	grpc_prometheus.Register(s.grpcServer)
+	if s.config.Metrics.Address != "" {
+		grpc_prometheus.Register(s.grpcServer)
```

Comment:
suggest logging (info level) to mention how the admin can enable prometheus (and histogram) metrics

Reply:
ok.thank you